File: 28_find_the_index_of_the_first_occurrence_in_a_string/strStr.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  let n = haystack.length;
  let m = needle.length;

  for (let i = 0; i < n - m + 1; i++) {
    if (haystack.slice(i, i + m) === needle) return i;
  }

  return -1;
};

File: 28_find_the_index_of_the_first_occurrence_in_a_string/str_str.rs
impl Solution {
    pub fn str_str(haystack: String, needle: String) -> i32 {
        let n = haystack.len();
        let m = needle.len();

        if m == 0 {
            return 0;
        }

        if n < m {
            return -1;
        }

        for i in 0..n-m + 1 {
            let mut j = 0;

            while j < m && haystack.chars().nth(i + j).unwrap() == needle.chars().nth(j).unwrap() {
                j+=1;
            }

            if j == m {
                return i as i32;
            }
        }

        -1
    }
}